x-pack/plugins/ml/public/settings/filter_lists/list/filter_lists.test.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+jest.mock('../../../privilege/check_privilege', () => ({
+  checkPermission: () => true
+}));
+
+// Mock the call for loading the list of filters.
+// The mock is hoisted to the top, so need to prefix the filter variable
+// with 'mock' so it can be used lazily.
+const mockTestFilter = {
+  filter_id: 'safe_domains',
+  description: 'List of known safe domains',
+  item_count: 500,
+  used_by: { jobs: ['dns_exfiltration'] },
+};
+jest.mock('../../../services/ml_api_service', () => ({
+  ml: {
+    filters: {
+      filtersStats: () => {
+        return Promise.resolve([mockTestFilter]);
+      }
+    }
+  }
+}));
+
+import { shallow } from 'enzyme';
+import React from 'react';
+
+import { FilterLists } from './filter_lists';
+
+describe('Filter Lists', () => {
+
+  test('renders a list of filters', () => {
+
+    const wrapper = shallow(
+      <FilterLists />
+    );
+
+    // Cannot find a way to generate the snapshot after the Promise in the mock ml.filters
```

Comment:
This article describes an approach where it's possible to access the promise from within the tests and put the tests inside `.then()` https://binarapps.com/blog/test-ajax-calls-in-react-component-lifecycle